hrjobcontract/js/src/job-contract/controllers/contract.js

```diff
@@ -69,8 +69,10 @@ define([
         _.extend($scope.health, newScope.health || contractRevisionIdObj);
         _.extend($scope.pension, newScope.pension || contractRevisionIdObj);
 
-        _.each($scope.leave, function (leaveType, leaveTypeId) {
-          _.extend(leaveType, newScope.leave ? newScope.leave[leaveTypeId] || contractRevisionIdObj : contractRevisionIdObj);
+        _.each($scope.leave, function (leaveType) {
+          _.extend(leaveType, newScope.leave
```

Comment:
put this whole block
```js
newScope.leave ? _.find(newScope.leave, { leave_type: leaveType.leave_type }) || contractRevisionIdObj : contractRevisionIdObj
```
in a separate function and document it

Reply:
done.